src/core/value.rs

```diff
@@ -116,6 +116,12 @@ impl Value {
         Value::Prod(Box::new(Value::u32(w0)), Box::new(Value::u32(w1)))
     }
 
+    pub fn u128(n: u128) -> Value {
+        let w0 = (n >> 64) as u64;
+        let w1 = (n & 0xffff_ffff_ffff_ffff) as u64;
```

Comment:
In c4e2f196c437980d3b5947839dbd20a8850bfa9a:

This masking isn't necessary. It doesn't really matter, but it does add a bit of review burden to have to count the `f`s.